datumaro/plugins/pointcloud_format/converter.py

```diff
@@ -0,0 +1,302 @@
+
+                for i, rimage in enumerate(item.related_images):
+
+                    try:
+                        name = rimage["name"]
+                    except AttributeError:
+                        name = "".join(random.choice(string.ascii_lowercase) for i in range(6))
+                        name = f"{name}.jpg"
+
+                    path = osp.join(related_dir, name)
+
+                    path = osp.abspath(path)
+                    os.makedirs(osp.dirname(path), exist_ok=True)
+
+                    if isinstance(rimage["image"], ByteImage):
+                        with open(path, 'wb') as f:
+                            f.write(item.get_bytes())
+                    else:
+                        save_image(path, rimage["image"].data)
+
+                    path = osp.join(related_dir, f"{name}.json")
+                    self._image_json["name"] = name
+                    with open(path, "w") as f:
+                        json.dump(self._image_json, f, indent=4)
+        else:
+            log.debug("Item '%s' has no image info", item.id)
+
+    def write_key_id_data(self, f):
+        json.dump(self._key_id_data, f, indent=4)
+
+    def write_meta_data(self, f):
+        json.dump(self._meta_data, f, indent=4)
+
+    def write_frame_data(self, f, key):
+        frame = {
+            "description": "",
+            "key": self.get_video_key(int(key)),
+            "tags": [],
+            "objects": self._label_objects,
+            "figures": {}
+        }
+
+        if self._frame_data.get(key):
+            frame["figures"] = self._frame_data[key]
+
+        json.dump(frame, f, indent=4)
+
+
+class PointCloudConverter(Converter):
+    DEFAULT_IMAGE_EXT = PointCloudPath.IMAGE_EXT
+
+    @classmethod
+    def build_cmdline_parser(cls, **kwargs):
+        parser = super().build_cmdline_parser(**kwargs)
+        parser.add_argument('--reindex', action='store_true',
+            help="Assign new indices to frames (default: %(default)s)")
+        parser.add_argument('--allow-undeclared-attrs', action='store_true',
+            help="Write annotation attributes even if they are not present in "
+                "the input dataset metainfo (default: %(default)s)")
+        return parser
+
+    def __init__(self, extractor, save_dir, reindex=False,
+            allow_undeclared_attrs=False, **kwargs):
+        super().__init__(extractor, save_dir, **kwargs)
+
+        self._reindex = reindex
+        self._builtin_attrs = PointCloudPath.BUILTIN_ATTRS
+        self._allow_undeclared_attrs = allow_undeclared_attrs
+
+    def apply(self):
+        self._default_dir = osp.join(self._save_dir, PointCloudPath.DEFAULT_DIR)
+        os.makedirs(self._default_dir, exist_ok=True)
+        self._annotation_dir = osp.join(self._default_dir, PointCloudPath.ANNNOTATION_DIR)
+        os.makedirs(self._annotation_dir, exist_ok=True)
+
+        for subset_name, subset in self._extractor.subsets().items():
```

Comment:
If this format does not support multiple subsets, use `self._extractor` itself, otherwise there is missing logic for this case.

Reply:
The format supports multiple subsets.